server/utils/applications/summaryListUtils.ts

```diff
@@ -55,38 +58,80 @@ const taskResponsesAsSummaryListItems = (
   return items
 }
 
-const attachDocumentsSummaryListItems = (
-  application: Application,
-  task: UiTask,
-  pageName: string,
-  showActions: boolean,
-) => {
+/**
+ * Return a list items from a document list for the check-your-answers page
+ * @param documents list of documents to render
+ * @param personCrn from application - used in document download path
+ * @param changePath - if defined, adds a 'change' link
+ * @param otherDocumentDetails - other documents text box contents
+ */
+export const getDocumentSummaryListItems = (
+  documents: Array<Document>,
+  personCrn: string,
+  changePath: string,
+  otherDocumentDetails: string,
+): Array<SummaryListItem> => {
   const items: Array<SummaryListItem> = []
 
-  documentsFromApplication(application).forEach(document => {
+  const getAction = (visuallyHiddenText = null as string) =>
```

Comment:
Could the type casting be avoided here?

```suggestion
  const getAction = (visuallyHiddenText = '') =>
```

Reply:
Fair enough. The only downside is that it would rely on checking for an empty string rather than null